### PR TITLE
remove dead code

### PR DESCRIPTION
**Pull request by deads2k**
This removes the dead code cruft since we stopped serving TPRs.

ref #48152

### CODE REVIEW

**Comment by deads2k** [comment]
federation race
/test pull-kubernetes-unit

File: pkg/apis/extensions/v1beta1/zz_generated.conversion.go [hunk 1]
```diff
@@ -346,7 +345,8 @@ func Convert_extensions_DaemonSetList_To_v1beta1_DaemonSetList(in *extensions.Da
 
 func autoConvert_v1beta1_DaemonSetSpec_To_extensions_DaemonSetSpec(in *v1beta1.DaemonSetSpec, out *extensions.DaemonSetSpec, s conversion.Scope) error {
 	out.Selector = (*v1.LabelSelector)(unsafe.Pointer(in.Selector))
-	if err := api_v1.Convert_v1_PodTemplateSpec_To_api_PodTemplateSpec(&in.Template, &out.Template, s); err != nil {
+	// TODO: Inefficient conversion - can we improve it?
+	if err := s.Convert(&in.Template, &out.Template, 0); err != nil {
```
Comment (sttts) [inline pkg/apis/extensions/v1beta1/zz_generated.conversion.go]: I also saw these kind of changes. Seem to be unrelated, but still strange.

File: pkg/kubectl/cmd/util/helpers.go [hunk 2]
```diff
@@ -676,18 +675,6 @@ func ParsePairs(pairArgs []string, pairType string, supportRemove bool) (newPair
 	return
 }
 
-// MaybeConvertObject attempts to convert an object to a specific group/version.  If the object is
-// a third party resource it is simply passed through.
-func MaybeConvertObject(obj runtime.Object, gv schema.GroupVersion, converter runtime.ObjectConvertor) (runtime.Object, error) {
```
Comment (sttts) [inline pkg/kubectl/cmd/util/helpers.go]: good that this goes away 👍

File: staging/src/k8s.io/apiserver/pkg/endpoints/watch_test.go [hunk 3]
```diff
@@ -272,7 +272,7 @@ func TestWatchRead(t *testing.T) {
 		// TODO: yaml stream serialization requires that RawExtension.MarshalJSON
 		// be able to understand nested encoding (since yaml calls json.Marshal
 		// rather than yaml.Marshal, which results in the raw bytes being in yaml).
-		// Same problem as thirdparty object.
+		// Same problem as customresource object.
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/endpoints/watch_test.go]: is this relevant anymore?

**Comment by sttts** [comment]
Some nits and a questionable conversion change.

File: staging/src/k8s.io/apiserver/pkg/endpoints/watch_test.go (hunk 3 above)
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/endpoints/watch_test.go]: > is this relevant anymore?

I don't know.

File: pkg/apis/extensions/v1beta1/zz_generated.conversion.go (hunk 1 above)
Comment (deads2k) [inline pkg/apis/extensions/v1beta1/zz_generated.conversion.go]: autocreated.  I'm not sure actually.  That does seem weird.

File: staging/src/k8s.io/apiserver/pkg/endpoints/watch_test.go (hunk 3 above)
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/endpoints/watch_test.go]: I got this as well in another PR, didn't commit it, continued working and suddenly it was gone. I fear this is something deep in the dockerized rsync build logic, somewhere between dragons, bash code and make fules.

File: pkg/apis/extensions/v1beta1/zz_generated.conversion.go (hunk 1 above)
Comment (sttts) [inline pkg/apis/extensions/v1beta1/zz_generated.conversion.go]: I got this as well in another PR, didn't commit it, continued working and suddenly it was gone. I fear this is something deep in the dockerized rsync build logic, somewhere between dragons, bash code and make fules.

**Comment by enisoc** [comment]
LGTM other than the mysterious unrelated conversion changes.

**Comment by deads2k** [comment]
Remove the generated file.

**Comment by deads2k** [comment]
apis was just dead validation cleanup.  approved.

**Pull request merged**